src/win/i2c.cc

```diff
@@ -0,0 +1,313 @@
+        wprintf(L"Slave address was not acknowledged\n");
+        break;
+      default:
+        wprintf(L"Invalid transfer status value\n");
+    }
+  }
+  catch (Exception^ e) {
+    return Nan::ThrowError(Nan::ErrnoException(e->HResult, "WinI2c::ReadPartial", ""));
+  }
+
+  for (int i = 0; i < readBuf->Length; i++) {
+    bufferData[i] = readBuf[i];
+  }
+
+  if (bytesRead == -1) {
+    return Nan::ThrowError(Nan::ErrnoException(errno, "WinI2c::ReadPartial", ""));
+  }
+
+  info.GetReturnValue().Set(Nan::New<v8::Integer>(bytesRead));
+}
+
+NAN_METHOD(WinI2c::Write) {
+  if (info.Length() < 1 || !info[0]->IsArray()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::Write",
+      "incorrect arguments passed to WinI2c::Write"));
+  }
+
+  std::vector<BYTE> bytes;
+  v8::Handle<v8::Value> val;
+  v8::Local<v8::Array> arr = Nan::New<v8::Array>();
+
+  v8::Handle<v8::Array> jsArray = v8::Handle<v8::Array>::Cast(info[0]);
+  for (unsigned int i = 0; i < jsArray->Length(); i++) {
+    val = jsArray->Get(i);
+    bytes.push_back(static_cast<BYTE>(val->Uint32Value()));
+    Nan::Set(arr, i, val);
+  }
+
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+  try {
+    obj->_i2cDevice->Write(ArrayReference<BYTE>(bytes.data(), static_cast<unsigned int>(bytes.size())));
+  } catch (Exception^ e) {
+    return Nan::ThrowError(Nan::ErrnoException(e->HResult, "WinI2c::Write", ""));
+  }
+}
+
+NAN_METHOD(WinI2c::WritePartial) {
+  if (info.Length() < 2 ||
+      !info[0]->IsInt32() ||
+      !info[1]->IsObject()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::WritePartial",
+      "incorrect arguments passed to WinI2c::WritePartial"
+      "(int length, Buffer buffer)"));
+  }
+
+  uint32 length = info[0]->Uint32Value();
+  v8::Local<v8::Object> bufferHandle = info[1].As<v8::Object>();
+  int bytesWritten = -1;
+  byte* bufferData = (byte*)node::Buffer::Data(bufferHandle);
+  size_t bufferLength = node::Buffer::Length(bufferHandle);
+
+  if (length > bufferLength) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::WritePartial",
+      "buffer passed to WinI2c::WritePartial contains less than 'length' bytes"));
+  }
+
+  auto writeBuf = ref new Platform::Array<BYTE>(length);
+
+  for (int i = 0; i < writeBuf->Length; i++) {
+    writeBuf->set(i, bufferData[i]);
+  }
+
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+  try {
+    I2cTransferResult result = obj->_i2cDevice->WritePartial(writeBuf);
+    switch (result.Status) {
+      case I2cTransferStatus::FullTransfer:
+        bytesWritten = result.BytesTransferred;
+        break;
+      case I2cTransferStatus::PartialTransfer:
+        bytesWritten = result.BytesTransferred;
+        wprintf(L"%d bytes partially transferred\n", bytesWritten);
+        break;
+      case I2cTransferStatus::SlaveAddressNotAcknowledged:
+        wprintf(L"Slave address was not acknowledged\n");
```

Comment:
Yes.

For example, running the following on Linux will return failure if the device at address 0x48 is not on the I2C bus:
```js
var i2c = require('../'),
  i2c1 = i2c.openSync(1);

var DS1621_ADDR = 0x48,
  CMD_ACCESS_CONFIG = 0xac;

i2c1.writeByteSync(DS1621_ADDR, CMD_ACCESS_CONFIG, 0x01);
```
There error is detected and thrown [here](https://github.com/fivdi/i2c-bus/blob/v1.0.3/src/writebyte.cc#L68-L70), and this is what ends up on the screen:
```
/home/pi/i2c-bus/node_modules/i2c-bus/i2c-bus.js:249
  i2c.writeByteSync(peripheralSync(this, addr), cmd, byte);
      ^

Error: EIO, Input/output error
    at Error (native)
    at Bus.writeByteSync (/home/pi/i2c-bus/node_modules/i2c-bus/i2c-bus.js:249:7)
    at Object.<anonymous> (/home/pi/i2c-bus/node_modules/i2c-bus/example/error.js:7:6)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:456:32)
    at tryModuleLoad (module.js:415:12)
    at Function.Module._load (module.js:407:3)
    at Function.Module.runMain (module.js:575:10)
    at startup (node.js:160:18)
```